docs/deployment/kubernetes/eks.md

```diff
@@ -0,0 +1,191 @@
+---
+title: EKS + Helm
+sidebar_position: 3
+---
+
+This guide will help you set up a Quickwit cluster on EKS with the correct S3 permissions.
+
+## Prerequisites
+- Running Elastic Kubernetes cluster (EKS)
+- `kubectl`
+- Permission to create the IAM role and Policies
+- AWS CLI
+- `eksctl` if you don't have an IAM OIDC provider for your cluster.
+
+## Set up
+
+Let's use the following environment variables:
+
+```bash
+export NAMESPACE=qw-tutorial
+export EKS_CLUSTER=qw-cluster
+export S3_BUCKET={your-bucket}
+export SERVICE_ACCOUNT_NAME=qw-sa
+export REGION={your-region}
+export CLUSTER_ID={your-cluster-id}
+```
+
+Create the namespace for our playground:
+
+```bash
+kubectl create ns ${NAMESPACE}
+```
+
+And set this namespace as the default one:
+
+```bash
+kubectl config set-context --current --namespace=${NAMESPACE}
+```
+
+
+### Create IAM OIDC provider if you don't have one
+
+To check if you have one provider for your EKS cluster, just run:
+
+```bash
+aws iam list-open-id-connect-providers
+```
+
+If you have one, you will get a response similar to this one:
+
+```json
+{
+    "OpenIDConnectProviderList": [
+        {
+            "Arn": "arn:aws:iam::(some-ID):oidc-provider/oidc.eks.{your-region}.amazonaws.com/id/{your-cluster-id}"
+        }
+    ]
+}
+```
+
+If you don't, run the following command:
+
+```bash
+eksctl utils associate-iam-oidc-provider --cluster ${EKS_CLUSTER} --approve
+```
+
+You can run again `aws iam list-open-id-connect-providers` to get the ARN of the provider.
+
+### Create an IAM policy
+
+You need to set the following policy to allow Quickwit to access your S3 bucket.
+
+Then create the policy using the AWS CLI:
+
+```bash
+cat > s3-policy.json <<EOF
+{
+    "Statement": [
+        {
+            "Action": [
+                "s3:AbortMultipartUpload",
+                "s3:DeleteObject",
+                "s3:GetObject",
+                "s3:ListMultipartUploadParts",
+                "s3:PutObject"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "arn:aws:s3:::${S3_BUCKET}/*"
+            ]
+        }
+    ],
+    "Version": "2012-10-17"
+}
+EOF
+```
+
+```bash
+aws iam create-policy --policy-name qw-s3-policy --policy-document file://s3-policy.json
+```
+
+### Create an IAM Role and attach the policy
+
+```bash
+cat > s3-role.json << EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${IAM_ID}:oidc-provider/oidc.eks.${REGION}.amazonaws.com/id/${CLUSTER_ID}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "oidc.eks.${REGION}.amazonaws.com/id/${CLUSTER_ID}:aud": "sts.amazonaws.com",
+          "oidc.eks.${REGION}.amazonaws.com/id/${CLUSTER_ID}:sub": "system:serviceaccount:${S3_BUCKET}:${SERVICE_ACCOUNT_NAME}"
```

Comment:
```suggestion
          "oidc.eks.${REGION}.amazonaws.com/id/${CLUSTER_ID}:sub": "system:serviceaccount:${NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
```

Think you meant to use NAMESPACE here rather than bucket